core/src/conn.rs

```diff
@@ -40,22 +167,19 @@ impl MongoConnection {
         operation_timeout: Option<u32>,
         login_timeout: Option<u32>,
         type_mode: TypeMode,
-        mut runtime: Option<tokio::runtime::Runtime>,
+        mut runtime: Option<Runtime>,
     ) -> Result<Self> {
-        let runtime = runtime.take().unwrap_or_else(|| {
+        let runtime = Arc::new(runtime.take().unwrap_or_else(|| {
             tokio::runtime::Builder::new_current_thread()
                 .enable_all()
                 .build()
                 .unwrap()
-        });
+        }));
         user_options.client_options.connect_timeout =
             login_timeout.map(|to| Duration::new(to as u64, 0));
-        let guard = runtime.enter();
-        let client = runtime.block_on(async {
-            Client::with_options(user_options.client_options).map_err(Error::InvalidClientOptions)
-        })?;
-        drop(guard);
         let uuid_repr = user_options.uuid_representation;
+        let (client, runtime) = Self::get_client_and_runtime(user_options, runtime)?;
+        // drop the Mutex as soon as possible
```

Comment:
Is this comment still valid, I don't see the mutex being dropped here unless it's referring to in `get_client_and_runtime()`